src/shogun2-core/src/main/java/de/terrestris/shogun2/service/PluginService.java

```diff
@@ -95,4 +146,14 @@ public String getPluginSource(String simpleClassName) throws Exception {
 	public void setDao(D dao) {
 		this.dao = dao;
 	}
+
+	/**
+	 * Sets the applicationService.
+	 *
+	 * @param applicationService
+	 */
+	@Autowired
```

Comment:
I dont like it, too, but we dont have another solution right now.

In future, we could try to:

* make the name of the qualifier configurable or
* do not scan more than one service in a hierarchy...